students/hirot_directory/session04/trigrams_hiro.py

```diff
@@ -0,0 +1,51 @@
+words = "I wish may I wish I might".split()
+
+print("The approach from class")
+
+trigrams = {}
+
+for i in range(len(words)-2):
+    pair = tuple(words[i:i+2])
+    follower = words[i+2]
+    print(pair, follower)
+    if pair in trigrams:
+        trigrams[pair].append(follower)
+    else:
+        trigrams[pair] = [follower] #key:pair
+
+print(trigrams)
+
+# .item :
+
+print("The approach with .item()")
+
+words = "I wish may I wish I might".split()
+
+trigrams = {}
+
+for i in range(len(words)-2):
+    pair = tuple(words[i:i+2])
+    follower = words[i+2]
+    print(pair, follower)
+    trigrams[pair] = follower
+
+print(trigrams.items())
+
```

Comment:
Hmm, since you are not using lists for your follower, you are losing some data. You have to replace that if block with something: Notice in the print statement from your dictionary above, "I wish" is followed once by "may" and once by 'I', but here you only get "I". The other change, printing as trigrams.items() rather than just printing the dictionary, really doesn't buy you much.